database/api/routes.md

```diff
@@ -0,0 +1,122 @@
+# api/v1/save/
+
+### Request Types(s)
++ POST
+
+### Request Body
+NOTE:: recon json key has been removed from previous versions
+```
+{
+    username: String
+    title: String
+    description: String
+    model: String
+    modelChecksum: String
+    graph: String
+    tags: [ String ]
+    scripts: [
+        { 
+            name: String
+            content: String
+            checksum: String
+        }
+    ]
+    files: [
+        {
+            name: String
+            uri: String
+            size: Integer
+            checksum: String
+            lastModified: String
+        }
+    ]
+    programBlock: [
+        {
+            id: Integer
+            inProgramBlock: Integer or NULL
+            name: String
+            qualifiedName: String
+        }
+    ]
+    data: [
+        {
+            id: Integer
+            inProgramBlock: Integer or NULL
+            name: String
+            qualifiedName: String
+        }
+    ]
+    port: [
+        {
+            id: Integer
+            onProgramBlock: Integer
+            data: Integer
+            name: String
+            qualifiedName: String
+            alias: String or NULL
+            uriTemplate: String or NULL
+            inPort: Bool
+            outPort: Bool
+        }
+    ]
+    channel: [
+        {
+            id: Integer
+            outPort: Integer
+            inPort: Integer
+            data: Integer
+            isInflow: Bool
+            isOutflow: Bool
+        }
+    ]
+    uriVariable: [ 
+        {
+            id: Integer
+            port: Integer
+            name: String
+        }
+    ]
+    resource: [ 
+        {
+            id: Integer
```

Comment:
Is it necessary to send the ID over JSON? Django can handle this for us and I can't think of a reason we would want to transfer IDs over

Reply:
Note: This comment applies to all IDs found in the JSON

Reply:
these arent the primary keys, they will be a part of a composite primary key with either `version_id` or `run_id`. check out the er diagram